### DISCUSSION TITLE
How can I display sensor data values in my Digital Twin?

### DISCUSSION THREAD

**Discussion by Telepresent** [iTwin/itwinjs-core#3436] [Q&A]
Hi, I would like to see IoT sensor data values in my Digital Twin. Do you have any sample?

Regards,

**Comment by ColinKerr** [comment]
Here is a code sample and a webinar that cover different ways to expose IoT data in an iTwin.

IoT Alerts sample: https://www.itwinjs.org/sandboxes/iTwinPlatform/IoT%20Alerts

iTwin Developer Webinar: Integrating and Visualizing BIM, IoT and Machine Learning: https://www.youtube.com/watch?v=zFL8pCm1UXA&list=PL6YCKeNfXXd9VFzTF-NSeN_l5vB7OTWR6&index=5

If you can provide more details on what you would like to do we can provide some more concrete suggestions.  For example if you want to add the data from IoT sensors to a widget in the viewer this tutorial shows you how to create a widget: https://developer.bentley.com/tutorials/itwin-viewer-hello-world/#2-your-first-ui-widget
